bedrock/utils/tests/test_views.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import unicode_literals
+
+from django.test import RequestFactory
+
+from bedrock.utils import views
+
+
+def test_variation_template_view():
+    rf = RequestFactory()
+    view = views.VariationTemplateView(template_context_variations=['b'],
+                                       template_name_variations=['b', 'c'],
+                                       template_name='mozorg/book.html')
+    req = rf.get('/', data={'v': 'b'})
+    view.request = req
+    assert view.get_context_data()['variation'] == 'b'
+    assert view.get_template_names() == ['mozorg/book-b.html']
+
+    req = rf.get('/', data={'v': 'dude'})
+    view.request = req
+    assert view.get_context_data()['variation'] == ''
+    assert view.get_template_names() == ['mozorg/book.html']
+
+    req = rf.get('/', data={'v': 'c'})
+    view.request = req
+    assert view.get_context_data()['variation'] == ''
+    assert view.get_template_names() == ['mozorg/book-c.html']
+
+    view = views.VariationTemplateView(template_name_variations=['1'],
+                                       template_name='about.index.html')
+    req = rf.get('/', data={'v': '1'})
+    view.request = req
+    assert 'variation' not in view.get_context_data()
+    assert view.get_template_names() == ['about.index-1.html']
```

Comment:
I didn't test that very much because I'm not sure they'll be useful to gether, but I also don't think it's necessary to prevent them from being used together. Since they rely on the same query param I don't think it'd be partiularly useful to use the context when you have a separate template.